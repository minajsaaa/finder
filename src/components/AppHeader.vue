<template lang="pug">
  div(class='header')
    div(class='header-inner')
      div(class='logo-container')
        router-link(to="/" exact): img(src="https://s3.ap-northeast-2.amazonaws.com/terra.money.home/static/finder/logo.svg")
      tm-form-struct(:submit="search")
        tm-form-group
          .tm-modal-search
            tm-field#search-input(
              type="text"
              placeholder="Search Block / Tx / Account"
              required
              v-model="query"
              autocomplete="off"
              title="1 to 60 characters")
            tm-btn(type="submit" icon="search")
</template>

<script>
import { mapGetters } from "vuex";
import TmFormGroup from "./TmFormGroup";
import TmFormStruct from "./TmFormStruct";
import TmBtn from "./TmBtn";
import TmField from "./TmField";
import { handleSearch } from "../scripts/utility";

export default {
  name: "app-header",
  components: {
    TmFormGroup,
    TmFormStruct,
    TmBtn,
    TmField
  },
  data: () => ({
    query: ``
  }),
  computed: {
    ...mapGetters(["config"])
  },
  methods: {
    search() {
      this.$router.push({
        path: handleSearch(this.query, this.$route.params.network)
      });
    }
  },
  mounted() {}
};
</script>

<style lang="stylus">
@require '../styles/variables'

.header
  position relative
  background #0C3694
  height 80px
  box-shadow 0 2px 4px 0 rgba(0, 0, 0, 0.1)

.header-inner
  width 100%
  margin 0 auto
  max-width 1440px
  position relative

.header .logo-container
  display inline-block

.header .logo-container a
  display block
  padding 28px 40px
  line-height 1

.header .logo-container img
  width 160px
  height 21px

.header .tm-form
  position absolute
  top 20px
  right 40px
  display inline-flex
  height 40px
  width 50%
  max-width 640px

.header .tm-form-group__field
  position relative

.header .tm-form-group
  padding 0

.header .tm-form-group input::-webkit-input-placeholder
  color rgba(255,255,255,.3)

.header .tm-form-group input:-moz-placeholder
  color rgba(255,255,255,.3)

.header .tm-form-group input::-moz-placeholder
  color rgba(255,255,255,.3)

.header .tm-form-group input:-ms-input-placeholder
  color rgba(255,255,255,.3)

.header .tm-form-group input
  border 0px
  border-bottom 1px solid rgba(255,255,255,.9)
  font-size 16px
  height 40px
  color #fff
  background transparent
  padding-left 10px !important
  padding-right 45px !important
  border-radius 0 !important

.header .tm-btn
  position absolute
  right: 0;
  bottom: 1px;
  height: 40px;
  outline: 0;

@media screen and (max-width: 767px)
  .header .logo-container
    display block

  .header
    height 44px

  .header .logo-container
    height 45px
    line-height 45px
    padding 0
    display flex
    flex-direction row
    justify-content center
    align-items center

  .header .logo-container img
    height 15px
    width 116px

  .header .logo-container a
    padding 13px

  .header .tm-form
    top 54px
    right 0
    padding 0
    width 100%
    padding: 0 15px
    max-width none

  .header .tm-form-group input
    border-radius 0 !important
    background #fff
    color #2043b5
    border-bottom : solid 1px rgba(32, 67, 181, .8);
    padding-left 5px !important
    padding-right 35px !important

  .header .tm-form-group input::-webkit-input-placeholder
    color: rgba(32, 67, 181,.3);

  .header .tm-form-group input:-moz-placeholder
    color: rgba(32, 67, 181,.3);

  .header .tm-form-group input::-moz-placeholder
    color: rgba(32, 67, 181,.3);

  .header .tm-form-group input:-ms-input-placeholder
    color: rgba(32, 67, 181,.3);
</style>
