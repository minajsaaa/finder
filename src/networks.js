const networks = [
  {
    value: "columbus-1",
    key: "Columbus-1",
    lcd: "https://lcd.terra.dev",
    rpc: "https://rpc.terra.dev",
    wss: "ws://rpc.terra.dev/websocket"
  },
  {
    value: "soju-0008",
    key: "Soju-0008 Testnet",
    lcd: "https://soju-0008-lcd.terra.dev",
    rpc: "https://soju-0008-rpc.terra.dev",
    wss: "ws://soju-0008-rpc.terra.dev/websocket"
  }
];

export const getNetwork = network =>
  networks.find(({ value }) => value === network) || {};

export default networks;
