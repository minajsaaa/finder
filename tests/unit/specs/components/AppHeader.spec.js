import { mount } from "@vue/test-utils";
import AppHeader from "@/components/AppHeader";
import networks from "@/networks";

describe(`AppHeader`, () => {
  let $router, $route, wrapper, localThis;
  const network = networks[0].value;

  beforeEach(() => {
    $router = {
      push: jest.fn()
    };

    $route = {
      params: {
        network
      }
    };
    wrapper = mount(AppHeader, {
      mocks: {
        $router,
        $route
      },
      stubs: [`router-link`]
    });
  });

  it(`AppHeader Snapshot`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  describe(`Tests using Localthis`, () => {
    beforeEach(() => {
      localThis = {
        $route: {
          params: {
            network
          }
        },
        $router: {
          push: jest.fn()
        },
        curNetwork: ``
      };
    });
    it(`mounted: set network`, () => {
      AppHeader.mounted.call(localThis);
      expect(localThis.curNetwork).toEqual(network);
    });

    it(`method: setNetwork`, () => {
      wrapper.find(".select-network select").setValue(network);
      expect(wrapper.vm.curNetwork).toEqual(network);
    });

    it(`method: search`, () => {
      wrapper.find("form").trigger("submit");
      expect(wrapper.vm.$router.push).toHaveBeenCalled();
    });
  });
});
