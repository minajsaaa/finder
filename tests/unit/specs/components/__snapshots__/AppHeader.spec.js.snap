// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader AppHeader Snapshot 1`] = `
<div
  class="header"
>
  <div
    class="header-inner"
  >
    <div
      class="logo-container"
    >
      <router-link-stub
        exact="exact"
        to="/"
      >
        <img
          src="https://s3.ap-northeast-2.amazonaws.com/terra.money.home/static/finder/logo.svg"
        />
      </router-link-stub>
    </div>
    <form
      class="tm-form"
    >
      <div
        class="tm-form-container"
      >
        <!---->
        <main
          class="tm-form-main"
        >
          <div
            class="tm-form-group"
          >
            <!---->
            <!---->
            <div
              class="tm-form-group__field"
            >
              <div
                class="tm-modal-search"
              >
                <input
                  autocomplete="off"
                  class="tm-field"
                  id="search-input"
                  placeholder="Search Block / Tx / Account"
                  required="required"
                  title="1 to 60 characters"
                  type="text"
                />
                <button
                  class="tm-btn"
                  type="submit"
                >
                  <span
                    class="tm-btn__container"
                  >
                    <i
                      aria-hidden="true"
                      class="tm-btn__icon material-icons"
                    >
                      search
                    </i>
                    <!---->
                    <!---->
                  </span>
                </button>
              </div>
            </div>
          </div>
        </main>
        <!---->
      </div>
    </form>
    <div
      class="select-network tm-form-group"
    >
      <!---->
      <!---->
      <div
        class="tm-form-group__field"
      >
        <div
          class="tm-select"
        >
          <select
            class="tm-field tm-field-select"
          >
            <option
              disabled="disabled"
              hidden="hidden"
              selected="selected"
              value=""
            >
              Select network...
            </option>
             
            <option
              value="columbus-1"
            >
              Columbus-1
            </option>
            <option
              value="soju-0008"
            >
              Soju-0008 Testnet
            </option>
            <option
              value="ricewine-0001"
            >
              Ricewine-0001 Testnet
            </option>
          </select>
           
          <div
            class="tm-field-select-addon"
          >
            <i
              class="material-icons"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
